### ISSUE TITLE
Extra headers added via round tripper break oauth2 exchange requests

### ISSUE THREAD

**Issue by csandanov** [microsoftgraph/msgraph-sdk-go#91]
I've spent a couple of days debugging my [oauth2 issue](https://stackoverflow.com/questions/71241676/random-errors-acquiring-microsoft-oauth2-token-via-golang-org-x-oauth2/71276835#71276835) only to find out that msgraph sdk implements a round tripper that adds extra headers that eventually break oauth2 exchange request. The round tripper calls middlewares including the compression handler from `microsoft/kiota` that adds `Content-Encoding: gzip` header that breaks oauth2 requests.

I'm not quite sure how exactly round tripper registration works but after calling these lines all of my subsequent oauth2 request break:

```go
import (
	a "github.com/microsoft/kiota/authentication/go/azure"
	msgraphsdk "github.com/microsoftgraph/msgraph-sdk-go"
)

auth, _ := a.NewAzureIdentityAuthenticationProvider(cred)
adapter, _ := msgraphsdk.NewGraphRequestAdapter(auth)
```

**Comment by baywet** [comment]
Hi @csandanov 
Thanks for trying the Go SDK and for reporting this issue.
If I understood things right, you're using a different authentication library in your application, and that auth library makes use of the default client.

Because the SDK is applying its configuration directly to the default client, this has negative impacts on other part of the application using the default client for other purposes.
https://github.com/microsoft/kiota/blob/c2dcdc98e868b5c467b294c7f182279a6d2d4cba/http/go/nethttp/kiota_client_factory.go#L18

I've authored a [first PR to fix this behaviour](https://github.com/microsoft/kiota/pull/1288), and I'll author subsequent PRs into the core library and here once it gets merged.

**Comment by csandanov** [comment]
Hi, and thank you for the fix. I'm using the standard oauth2 library `golang.org/x/oauth2`, it has this behaviour of using the default http client if a client not specified in the context (my current workaround)

**Comment by baywet** [comment]
update: the pr in kiota has been merged, a PR in core has been created, merged and released, now opened #93 to fix this issue.
